Source/WebCore/rendering/svg/SVGResources.cpp

```diff
@@ -307,22 +307,38 @@ std::unique_ptr<SVGResources> SVGResources::buildCachedResources(const RenderEle
     return foundResources;
 }
 
-void SVGResources::layoutDifferentRootIfNeeded(const LegacyRenderSVGRoot* svgRoot)
+void SVGResources::layoutDifferentRootIfNeeded(const RenderElement& resourcesClient)
 {
+    const LegacyRenderSVGRoot* clientSVGRoot = nullptr;
+
     auto layoutDifferentRootIfNeeded = [&](RenderElement* container) {
         if (!container)
             return;
+
         auto* root = SVGRenderSupport::findTreeRootObject(*container);
-        if (svgRoot == root || root->isInLayout())
+        if (root->isInLayout())
+            return;
+
+        if (!clientSVGRoot)
+            clientSVGRoot = SVGRenderSupport::findTreeRootObject(resourcesClient);
+
+        if (clientSVGRoot == root)
             return;
+
         container->layoutIfNeeded();
     };
-    layoutDifferentRootIfNeeded(clipper());
-    layoutDifferentRootIfNeeded(masker());
-    layoutDifferentRootIfNeeded(filter());
-    layoutDifferentRootIfNeeded(markerStart());
-    layoutDifferentRootIfNeeded(markerMid());
-    layoutDifferentRootIfNeeded(markerEnd());
+
+    if (m_clipperFilterMaskerData) {
+        layoutDifferentRootIfNeeded(clipper());
```

Comment:
This will save another check for `m_clipperFilterMaskerData` in `SVGResources::clipper()`

```
layoutDifferentRootIfNeeded(m_clipperFilterMaskerData->clipper.get());
```